### PR TITLE
web-site content; why-netdata content

### PR DESCRIPTION
**Pull request by ktsaou**
##### Summary

1. New content at the site
2. Updated "Why Netdata" content

### CODE REVIEW

File: docs/why-netdata/1s-granularity.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+# 1s granularity
+
+High resolution metrics are required to effectively monitor and troubleshoot
+systems and applications.
+
+## why?
```
Comment (gmosx) [inline docs/why-netdata/1s-granularity.md]: The subsection titles should start with a capital letter.

File: docs/why-netdata/1s-granularity.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+- The world is going real-time. Today, customers' experience is significantly affected by response time, so SLAs are tighten. It is just not practical to monitor a 2-second SLA with 10-second metrics.
+
+- IT goes virtual. Unlike real hardware, virtual environments are not predictable. You cannot expect resources to be there when your applications need them. They will eventually be, but not exactly at the time they are needed. The latency of virtual environments is affected by many factors, most of which are outside our control, like: the maintenance policy of the hosting provider, the work load of third party virtual machines running on the same physical servers combined with the resource allocation and throttling policy among virtual machines, the provisioning system of the hosting provider, etc.
```
Comment (gmosx) [inline docs/why-netdata/1s-granularity.md]: `resources to be there` -> `resources to be available`.

File: docs/why-netdata/1s-granularity.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+Of course, Netdata can centralize metrics when needed. For example, it is not
+practical to keep metrics locally on ephemeral nodes. For these cases, Netdata
+streams the metrics in real-time, from the ephemeral nodes to one or more
+non-ephemeral nodes nearby. This centralization is again distributed. On a large
+infrastructure there may be many centralization points.
+
+To eliminate the error introduced by data collection latencies on busy virtual
+environments, Netdata interpolates collected metrics. It does this using
+microsecond timings, per data source, offering measurements with an error rate
+of 0.00001%.
```
Comment (paulfantom) [inline docs/why-netdata/1s-granularity.md]: Would be nice to have something backing up this claim.

File: docs/why-netdata/README.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+Netdata is built around 4 principles:
+
+1. **[Per second data collection for all metrics.](1s-granularity.md)**
+
+    *It is impossible to monitor a 2 second SLA, with 10 second metrics.*
+
+2. **[Collect and visualize all the metrics from all possible sources.](unlimited-metrics.md)**
+
+    *To troubleshoot slowdowns, we need all the available metrics. The console should provide nothing more.*
```
Comment (gmosx) [inline docs/why-netdata/README.md]: `The console should provide nothing more` <- what does this mean?

File: docs/why-netdata/1s-granularity.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+Finally, Netdata is really fast. Optimization is a core product feature.
+On modern hardware, Netdata can collect metrics with a rate of above 1M metrics
```
Comment (paulfantom) [inline docs/why-netdata/1s-granularity.md]: Very ambiguous: what is "modern hardware", where can someone read about how that test was performed or how can someone test it by himself?

File: docs/why-netdata/README.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+3. **[Meaningful presentation, optimized for visual anomaly detection.](meaningful-presentation.md)**
+
+    *Metrics are a lot more than name-value pairs over time. The monitoring tool should know all the metrics. We should not!*
```
Comment (gmosx) [inline docs/why-netdata/README.md]: `We should not` -> `Users should not`

File: docs/why-netdata/1s-granularity.md (hunk 3 above)
Comment (ktsaou) [reply docs/why-netdata/1s-granularity.md]: You mean about the 100ms latency or the netdata accuracy?

100ms latency, randomly spread over all proc data collection: https://docs.google.com/presentation/d/18C8bCTbtgKDWqPa57GXIjB2PbjjpjsUNkLtZEz6YK8s/edit#slide=id.g422e696d87_0_57

Netdata measures its accuracy, when in debug mode: https://github.com/netdata/netdata/blob/36199f449852f8077ea915a3a14a33fa2aff6d85/database/rrdset.c#L1070-L1099

So, it checks every point and also the total volume of collected metrics, and logs if accuracy goes above 0.0001% (3 zeros, not 4 - my mistake): https://github.com/netdata/netdata/blob/36199f449852f8077ea915a3a14a33fa2aff6d85/libnetdata/storage_number/storage_number.h#L85

I'll add more info to the document.

File: docs/why-netdata/1s-granularity.md (hunk 5 above)
Comment (ktsaou) [reply docs/why-netdata/1s-granularity.md]: Modern Hardware = anything you can get on AWS, GC, Azure. The smallest VM offered is modern enough...

File: docs/why-netdata/1s-granularity.md (hunk 1 above)
Comment (ktsaou) [reply docs/why-netdata/1s-granularity.md]: yeap

File: docs/why-netdata/1s-granularity.md (hunk 2 above)
Comment (ktsaou) [reply docs/why-netdata/1s-granularity.md]: sure

File: docs/why-netdata/README.md (hunk 4 above)
Comment (ktsaou) [reply docs/why-netdata/README.md]: "The console should not provide more metrics"
fixed it

File: docs/why-netdata/README.md (hunk 6 above)
Comment (ktsaou) [reply docs/why-netdata/README.md]: sure

File: docs/why-netdata/unlimited-metrics.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+Collecting all the metrics breaks the first rule of every monitoring text book: "collect only the metrics you need", "collect only the metrics you understand".
+
+Unfortunately, this does not work! Filtering out most metrics is like reading a book by skipping most of its pages...
+
+For many people, monitoring is about:
+
+- Detecting outages
+- Capacity planning
+
+However, today **slowdowns are 10 times more common** compared to outages.
```
[the remark this answers is not in the document]
Comment (ktsaou) [reply docs/why-netdata/unlimited-metrics.md]: slide 14 of https://www.slideshare.net/KenGodskind/alertsitetrac

Slowdowns are more expensive than outages: https://dzone.com/articles/downtime-vs-slowtime-which-hurts-more

I also validate this from experience. During the last few years as the COO of an online business residing in the cloud, I dealt with just a bunch of outages and a many hundreds of slowdowns. Especially for online businesses running transactions on cloud infrastructure, slowdowns are a daily issue and concern. Slowdowns are unpredictable, they come and go randomly, they are the harder to trace.

File: docs/why-netdata/unlimited-metrics.md [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+
+## what others do?
+
+So, why monitoring solutions and SaaS providers filter out metrics?
+
+Well... they can't do otherwise!
+
+1. Time-series databases limit the number of metrics collected, because the number of metrics influences their performance significantly. They get congested at scale.
+
+3. SaaS providers centralize all the metrics so they depend on metrics filtering to control their costs.
```
[the remark this answers is not in the document]
Comment (ktsaou) [reply docs/why-netdata/unlimited-metrics.md]: sure, fixed

**Comment by cakrit** [comment]
Are "Meaningful Presentation" and "Immediate Results" left empty on purpose for now?

File: docs/why-netdata/1s-granularity.md (hunk 5 above)
Comment (ktsaou) [reply docs/why-netdata/1s-granularity.md]: added a description of "modern hardware"

**Comment by ktsaou** [comment]
> Are "Meaningful Presentation" and "Immediate Results" left empty on purpose for now?

I'll write them, today I hope.

File: docs/why-netdata/unlimited-metrics.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (paulfantom) [reply docs/why-netdata/unlimited-metrics.md]: So link to it in text.

[the remark this answers is not in the document]
Comment (ktsaou) [reply docs/why-netdata/unlimited-metrics.md]: added them

**Comment by ktsaou** [comment]
ok, done!
Please review to have this merged and update the web-site.

**Comment by ktsaou** [comment]
I merged it to update the site, but please make additional PRs to improve this.

**Pull request merged**